consensus/spos/bls/subroundEndRound.go

```diff
@@ -553,6 +565,119 @@ func (sr *subroundEndRound) aggregateSigsAndHandleInvalidSigners(bitmap []byte)
 	return bitmap, sig, nil
 }
 
+func (sr *subroundEndRound) checkGoRoutinesThrottler(ctx context.Context) error {
+	for {
+		if sr.signatureThrottler.CanProcess() {
+			break
+		}
+		select {
+		case <-time.After(time.Millisecond):
+			continue
+		case <-ctx.Done():
+			return spos.ErrTimeIsOut
+		}
+	}
+	return nil
+}
+
+func (sr *subroundEndRound) signatureVerification(wg *sync.WaitGroup, i int, pk string, invalidPubKey *[]string, mutex *sync.Mutex, sigShare []byte, mutexBool *sync.Mutex, errorReturned *error) {
+	defer wg.Done()
+
+	isSuccessful := true
+	err := sr.SigningHandler().VerifySignatureShare(uint16(i), sigShare, sr.GetData(), sr.Header.GetEpoch())
+	if err != nil {
+		isSuccessful = false
+		err = sr.SetJobDone(pk, SrSignature, false)
+		if err != nil {
+			if *errorReturned == nil {
+				mutexBool.Lock()
+				*errorReturned = err
+				mutexBool.Unlock()
+			}
+			return
+		}
+		decreaseFactor := -spos.ValidatorPeerHonestyIncreaseFactor + spos.ValidatorPeerHonestyDecreaseFactor
+		sr.PeerHonestyHandler().ChangeScore(
+			pk,
+			spos.GetConsensusTopicID(sr.ShardCoordinator()),
+			decreaseFactor,
+		)
+		mutex.Lock()
+		*invalidPubKey = append(*invalidPubKey, pk)
+		mutex.Unlock()
+	}
+
+	log.Trace("verifyNodesOnAggSigVerificationFail: verifying signature share", "public key", pk, "is successful", isSuccessful)
+}
+
+func (sr *subroundEndRound) verifyNodesOnAggSigFailAux() ([]string, error) {
+	invalidPubKeys := make([]string, 0)
+	pubKeys := sr.ConsensusGroup()
+	wg := &sync.WaitGroup{}
+	mutex := &sync.Mutex{}
+	mutexBool := &sync.Mutex{}
+	var errorReturned error = nil
+	if check.IfNil(sr.Header) {
+		return nil, spos.ErrNilHeader
+	}
+	for i, pk := range pubKeys {
+		isJobDone, err := sr.JobDone(pk, SrSignature)
+		if err != nil || !isJobDone {
+			continue
+		}
+		sigShare, err := sr.SigningHandler().SignatureShare(uint16(i))
+		if err != nil {
+			return nil, err
+		}
+		wg.Add(1)
+		go sr.signatureVerification(wg, i, pk, &invalidPubKeys, mutex, sigShare, mutexBool, &errorReturned)
+	}
+	wg.Wait()
+	if errorReturned != nil {
+		return nil, errorReturned
+	}
+	return invalidPubKeys, nil
+}
+
+func (sr *subroundEndRound) verifyNodesOnAggSigFailAuxThrottle(ctx context.Context) ([]string, error) {
```

Comment:
we should keep only the version with throttle, but give it the name `verifyNodesOnAggSigFail`